clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -2884,6 +2884,12 @@ def warn_cxx11_compat_constexpr_body_multiple_return : Warning<
   InGroup<CXXPre14Compat>, DefaultIgnore;
 def note_constexpr_body_previous_return : Note<
   "previous return statement is here">;
+def err_ms_constexpr_not_distinct : Error<
+  "[[msvc::constexpr]] cannot be applied to a %select{constexpr|consteval}0 function %1">;
+def err_ms_constexpr_virtual : Error<"virtual function cannot be [[msvc::constexpr]]">;
```

Comment:
This diagnostic isn't particularly well worded, and, IMO, would be suited by just adding `virtual` to the select in the diagnostic on 2887.

Reply:
Done.